unidist/core/base/utils.py

```diff
@@ -4,27 +4,42 @@
 
 """Utilities used to initialize execution backend."""
 
-import os
-
+from unidist.config import Backend, CpuCount
 from .backend import BackendProxy
 
 
-def init_backend():
+def init_backend(backend=None, num_cpus=None):
     """
     Initialize an execution backend.
 
+    Parameters
+    ----------
+    backend : str, optional
+        Backend that unidist should run on.
+    num_cpus : int, optional
+        Number of CPUs that should be used by backend.
+
     Notes
     -----
-    The concrete execution backend is chosen in depend on
-    `UNIDIST_BACKEND` environment variable.
-    If the variable is not set, Ray backend is used.
+    The concrete execution backend can also be chosen via
+    `UNIDIST_BACKEND` environment variable or ``Backend`` config value.
+    Ray backend is used by default.
     """
-    backend_name = os.environ.get("UNIDIST_BACKEND", "Ray")
+    if num_cpus is not None:
+        CpuCount.put(num_cpus)
+    else:
+        num_cpus = CpuCount.get()
+
+    if backend is not None:
+        Backend.put(backend)
+    else:
+        backend_name = Backend.get()
+
```

Comment:
`backend_name` is set only in case `backend` is None. In the result:
```python
In [1]: import unidist as ud
In [2]: ud.init(backend="MPI")
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-2-c4164c81a520> in <module>
----> 1 ud.init(backend="MPI")

/localdisk/aprutsko/unidist/unidist/api.py in init(backend, num_cpus)
     97     Ray backend is used by default.
     98     """
---> 99     init_backend(backend=backend, num_cpus=num_cpus)
    100
    101

/localdisk/aprutsko/unidist/unidist/core/base/utils.py in init_backend(backend, num_cpus)
     36         backend_name = Backend.get()
     37
---> 38     if backend_name == "Ray":
     39         from unidist.core.backends.ray.backend import RayBackend
     40         from unidist.core.backends.ray.utils import initialize_ray

UnboundLocalError: local variable 'backend_name' referenced before assignment
```

It could be simply handled using approach described here https://github.com/modin-project/unidist/pull/21#discussion_r751118881

Reply:
See the answer there.